packages/form-js-editor/src/render/components/FieldResizer.js

```diff
@@ -0,0 +1,188 @@
+import {
+  useRef
+} from 'preact/hooks';
+
+import {
+  classes,
+  query as domQuery
+} from 'min-dom';
+
+import { useService } from '../hooks';
+
+import { createDragger, throttle } from './Util';
+
+import { DRAG_NO_MOVE_CLS } from '../../features/dragging/Dragging';
+
+import classNames from 'classnames';
+
+const COLUMNS_REGEX = /^cds--col(-lg)?/;
+
+const ELEMENT_RESIZING_CLS = 'fjs-element-resizing';
+
+const RESIZE_DRAG_PREVIEW_CLS = 'fjs-resize-drag-preview';
+
+const GRID_OFFSET_PX = 16;
+
+
+export function FieldResizer(props) {
+
+  const {
+    field,
+    position
+  } = props;
+
+  const ref = useRef(null);
+
+  const formLayoutValidator = useService('formLayoutValidator');
+  const modeling = useService('modeling');
+
+  // we can't use state as we need to
+  // manipulate this inside dragging events
+  const context = useRef({
+    startColumns: null,
+    newColumns: null
+  });
+
+  const onResize = throttle((_, delta) => {
+    const { x: dx } = delta;
+
+    const { layout = {} } = field;
+
+    const newColumns = calculateNewColumns(
+      ref.current,
+      layout.columns || context.current.startColumns,
+      dx,
+      position
+    );
+
+    const errorMessage = formLayoutValidator.validateField(field, newColumns);
+
+    if (!errorMessage) {
+
+      context.current.newColumns = newColumns;
+
+      // make visual updates to preview change
+      const columnNode = ref.current.closest('.fjs-layout-column');
+      removeMatching(columnNode, COLUMNS_REGEX);
+      columnNode.classList.add(`cds--col-lg-${newColumns}`);
+    }
+  });
+
+  const onResizeStart = (event) => {
+
+    const target = getElementNode(field);
+    const parent = getParent(target);
+
+    // create a blank element to use as drag preview
+    // ensure it was only created once
+    let blankPreview = getDragPreviewImage(parent);
+
+    if (!blankPreview) {
+      blankPreview = document.createElement('div');
+      blankPreview.classList.add(RESIZE_DRAG_PREVIEW_CLS);
+      parent.appendChild(blankPreview);
+    }
+
+    // initialize drag handler
+    const onDragStart = createDragger(onResize, blankPreview);
+    onDragStart(event);
+
+    // mitigate auto columns on the grid that
+    // has a offset of 16px (1rem) to both side
+    const columnNode = getColumnNode(target);
+    const startWidth = columnNode.getBoundingClientRect().width + GRID_OFFSET_PX;
+    context.current.startColumns = asColumns(startWidth, parent);
+
+    setResizing(target, position);
+  };
+
+  const onResizeEnd = () => {
+    const { layout = {} } = field;
+
+    if (context.current.newColumns) {
+      modeling.editFormField(field, 'layout', {
+        ...layout,
+        columns: parseInt(context.current.newColumns)
```

Comment:
is `parseInt` really necessary? it seems that we always only deal with numbers with the context.

Reply:
In this case I think you are right, likely no strings ever involved.